registry/storage/catalog.go

```diff
@@ -49,17 +48,22 @@ func (reg *registry) Repositories(ctx context.Context, repos []string, last stri
 
 		// if we've filled our array, no need to walk any further
 		if len(foundRepos) == len(repos) {
```

Comment:
Yes, there could be one additional `foundRepos` entry that isn't copied, but the argument `last` we use to skip repositories is computed from the `repos` buffer we passed in, so we won't skip any repositories on a subsequent call. Unfortunately, there isn't a way without changing `Walk` to lookahead without letting it incur the extra `Stat` on the next child.